qiskit_experiments/library/characterization/drag.py

```diff
@@ -133,63 +121,73 @@ def set_experiment_options(self, reps: Optional[List] = None, **fields):
+        betas: Optional[Iterable[float]] = None,
+        backend: Optional[Backend] = None,
+    ):
+        """Initialize a Drag experiment in the given qubit.
+
         Args:
             qubit: The qubit for which to run the Drag calibration.
+            schedule: The schedule to run. This schedule should have one free parameter
+                corresponding to a DRAG parameter.
+            betas: The values of the DRAG parameter to scan. If None is given the default range
+                :code:`linspace(-5, 5, 51)` is used.
             backend: Optional, the backend to run the experiment on.
+
+        Raises:
+            QiskitError: if the schedule does not have a free parameter.
         """
 
         super().__init__([qubit], backend=backend)
 
+        if betas is not None:
+            self.experiment_options.betas = betas
+
+        if len(schedule.parameters) != 1:
+            raise QiskitError(
+                f"Schedule {schedule} for {self.__class__.__name__} experiment must have "
+                f"exactly one free parameter, found {schedule.parameters} parameters."
+            )
+
+        self.experiment_options.schedule = schedule
+
+    def _pre_circuit(self) -> QuantumCircuit:
+        """A circuit with operations to perform before the Drag."""
+        return QuantumCircuit(1)
+
     def circuits(self) -> List[QuantumCircuit]:
         """Create the circuits for the Drag calibration.
 
         Returns:
             circuits: The circuits that will run the Drag calibration.
 
         Raises:
-            CalibrationError:
-                - If the beta parameters in the xp and xm pulses are not the same.
-                - If either the xp or xm pulse do not have at least one Drag pulse.
-                - If the number of different repetition series is not three.
+            QiskitError: if the number of different repetition series is not three.
         """
         schedule = self.experiment_options.schedule
 
-        if schedule is None:
-            beta = Parameter("β")
-            with pulse.build(backend=self.backend, name="drag") as schedule:
-                pulse.play(
-                    pulse.Drag(
-                        duration=self.experiment_options.duration,
-                        amp=self.experiment_options.amp,
-                        sigma=self.experiment_options.sigma,
-                        beta=beta,
-                    ),
-                    pulse.DriveChannel(self.physical_qubits[0]),
-                )
-
-        if len(schedule.parameters) != 1:
-            raise CalibrationError(
-                "The schedule for Drag calibration must have one free parameter."
-                f"Found {len(schedule.parameters)}."
-            )
-
         beta = next(iter(schedule.parameters))
 
-        drag_gate = Gate(name=schedule.name, num_qubits=1, params=[beta])
+        # Note: if the pulse has a reserved name, e.g. x, which does not have parameters
```

Comment:
This sounds bit strange to me. Indeed you can
```python
    x = Gate(name="x", num_qubits=1, params=[beta])
    
    circuit.append(drag_gate, (0,))
    circuit.rz(np.pi, 0)
    circuit.append(x, (0,))
    circuit.rz(np.pi, 0)
    circuit.add_calibration("x", (0,), schedule)
```
I don't think this is not naming issue (as long as you attach cal with the same name), but just you need to define a gate with parameters (because transpiler doesn't check parameters within the schedule).

Reply:
I can confirm that this bit of code:
```python
        drag_gate = Gate(name=schedule.name, num_qubits=1, params=[beta])

        reps = self.experiment_options.reps
        if len(reps) != 3:
            raise QiskitError(
                f"{self.__class__.__name__} uses exactly three repetitions. "
                f"Received {reps} with length {len(reps)} != 3."
            )

        circuits = []

        for idx, rep in enumerate(reps):
            circuit = self._pre_circuit()
            for _ in range(rep):
                circuit.append(drag_gate, (0,))
                circuit.rz(np.pi, 0)
                circuit.append(drag_gate, (0,))
                circuit.rz(np.pi, 0)

            circuit.measure_active()

            circuit.add_calibration(schedule.name, self.physical_qubits, schedule, params=[beta])
```
does not work when the schedule is called `x`. It results in the following error:
```
failed: 
qiskit.providers.ibmq.job.exceptions.IBMQJobFailureError: 
\'Unable to retrieve result for job 618390695fd44a5545c22a1d. 
Job has failed: QObj does not conform to the schema. Error code: 1101.
```
However, when the gate is name `Drag(x)` it works fine. My suspicion is that `x` is a reserved name or something and this fails some form of QObj validation. I'm going to leave the code with 
```python
drag_gate = Gate(name="Drag(" + schedule.name + ")", num_qubits=1, params=[beta])
```
since this is safer and runs.

Reply:
I see. I'm fine with your code but it should be reported to backend team properly.